### ISSUE TITLE
0.21.3 leaks linter tests to containing app

### ISSUE THREAD

**Issue by pzuraq** [ef4/liquid-fire#370]
The latest version of liquid-fire is leaking it's linter tests to our app and causing failures in our tests. Weirdly, this only happens on one of our CI environments. Travis is fine.

**Ember-CLI:** 1.13.1
**Ember:** 1.13.4
**Ember Data:** 1.0.0-beta.19.1
**jQuery:** 2.1.4

**Comment by olivierlesnicki** [comment]
+1

**Comment by olivierlesnicki** [comment]
downgrading to `0.21.2` removed the issue.
Its is a `0.21.3` specific issue

**Comment by ef4** [comment]
The only change I can see between 0.21.2 and 0.21.3 that seems plausibly responsible is that liquid-fire's own ember-cli-qunit version bumped from 0.3.19 to 1.0.3. Hmm...

**Comment by jrjohnson** [comment]
I was seeing the error
```
modules/liquid-fire/tabbable.js should pass jshint.
modules/liquid-fire/tabbable.js: line 15, col 5, Redefinition of '$'.
```
Which I think is now fixed with PR #371.  

Is that what everyone else is seeing?  I'm not sure why JSHint suddenly started noticing this as I have had a $ global for a while and the tabbable file hasn't been touched recently.

**Comment by ef4** [comment]
I am unable to reproduce this using ember-cli-1.13.8. Please reopen if you can.

**Comment by jrjohnson** [comment]
I'm seeing this in 1.13.8.  Should be reproducible by adding $ to the predef section of .jshint
```
"predef": [
    "document",
    "window",
    "-Promise",
    "$"
  ],
```

**Comment by ef4** [comment]
I tried that, can't reproduce.

Please check your versions of ember-cli-qunit and ember-qunit.

**Comment by jrjohnson** [comment]
I'm up to date with both `"ember-cli-qunit": "^1.0.0"` and `"ember-qunit": "0.4.9"`.  It looks like there are new versions in ember-cli-master I will try those and see if that changes anything.

**Comment by pzuraq** [comment]
Note that the issue was seen in our local Jenkins build and not in Travis, so this is probably a problem fairly specific to our setup. Going back through the logs it looks like we were getting a lot of jscs errors specifically.

Things like:
```
Variable declarations should use `let` or `const` not `var` at modules/liquid-fire/modal.js :
```

**Comment by ef4** [comment]
@jrjohnson Are you using ember-sauve or JSCS?

**Comment by jrjohnson** [comment]
No, just the default jshint rules and qunit tests provided with ember-cli.  This is what a test run looks like clipped for brevity:
```bash
$ ember t --filter='JSHint'
version: 1.13.8
ok 1 PhantomJS 2.0 - JSHint - modules/el-calendar/components: modules/el-calendar/components/calendar-event.js should pass jshint
ok 11 PhantomJS 2.0 - JSHint - modules/ilios-calendar/components: modules/ilios-calendar/components/ilios-calendar-day.js should pass jshint
ok 17 PhantomJS 2.0 - JSHint - modules/liquid-fire: modules/liquid-fire/action.js should pass jshint
```
These three lines all represent addons that I have installed.  They are the only addons that appear in the list of JSHint tests so they must have something in-common and I'm trying to figure out what.

Of note liquid-fire 0.21.2 will not show up in the list if it is installed.  Just v0.21.3.

**Comment by elwayman02** [comment]
@ef4 I can reproduce this for JSCS tests:

liquid-fire 0.21.3
ember-suave 1.1.0
ember-cli-qunit ~1.0.3
ember-qunit 0.4.13
qunit ~1.19.0

**Comment by ivanvotti** [comment]
The same here. I have a lot of JSCS and some JSHint broken tests from  `modules/liquid-fire`
```
"liquid-fire": "^0.21.3",
"ember-cli-qunit": "^1.0.0",
"ember-qunit": "0.4.9",
"qunit": "~1.18.0",
"broccoli-jscs": "^1.1.0",
```

**Comment by xkb** [comment]
![image](https://cloud.githubusercontent.com/assets/5749667/10438457/e3760d50-70f0-11e5-8405-a99120c23ebd.png)

Using:
```
"ember-cli-qunit": "v1.0.3",
"liquid-fire": "^0.21.3"
"ember-qunit": "0.4.13",
"qunit": "~1.19.0",
```

I can confirm that this isn't present on `0.21.2`

**Comment by ivanvotti** [comment]
@ef4 Let's reopen this issue? Looks like many people can reproduce it with `liquid-fire 0.21.3`

**Comment by ascot21** [comment]
Weird. If I install liquid-fire by putting the version number in my package.js file like `"liquid-fire": "0.21.3"
`, the tests leak like others are seeing.

Yet, if I install it by linking to github with `"liquid-fire": "ef4/liquid-fire"` the tests don't leak into mine and everything works fine.

Using:
```
liquid-fire 0.21.3
ember-suave 1.1.0
ember-cli-qunit ~1.0.3
ember-qunit 0.4.13
qunit ~1.19.0
```

@ef4 is it possible the release to npm is messed up somehow?

**Comment by Samsinite** [comment]
@ef, I see a PR in for this issue, anything we can do to help it get accepted or offer another solution?

**Comment by mfenniak** [comment]
I can corroborate @ascot21's findings.  If I include liquid-fire in my package.json as "0.21.3", then I start getting errors showing my linter (jshint) rules running and failing against liquid-fire.  On the other hand, if I refer to liquid-fire by it's GitHub repo & tag ("ef4/liquid-fire#v0.21.3"), then jshint doesn't run against liquid-fire's code.

**Comment by spruce** [comment]
I'd like to have this fix to. :+1:

**Comment by winne42** [comment]
> @ef4 is it possible the release to npm is messed up somehow?
@ef4  Could this be related to https://github.com/ember-animation/liquid-fire/issues/383 ?
